lib/core/src/chain_swap.rs

```diff
@@ -394,10 +402,14 @@ impl ChainSwapHandler {
                 debug!("Zero-amount swap validated. Auto-accepting...");
                 self.persister
                     .update_actual_payer_amount(&id, user_lockup_amount_sat)?;
-                self.swapper
-                    .accept_zero_amount_chain_swap_quote(&id, quote)?;
                 self.persister
-                    .update_accepted_receiver_amount(&id, receiver_amount_sat)
+                    .update_accepted_receiver_amount(&id, Some(receiver_amount_sat))?;
+                self.swapper
+                    .accept_zero_amount_chain_swap_quote(&id, quote)
+                    .inspect_err(|e| {
+                        error!("Failed to accept zero-amount swap {id} quote: {e} - trying to erase the accepted receiver amount...");
+                        let _ = self.persister.update_accepted_receiver_amount(&id, None);
```

Comment:
Question is do we need that (reverting the accepted amount)?
We could have cases where we won't be able to revert, just wanted to ensure we handle this case gracefully

Reply:
> Question is do we need that (reverting the accepted amount)?

Yes, a payment that had its fees accepted (`accepted_receiver_amount_sat` is `Some`) will not allow fetching/ accepting new fees, and will be `Pending` instead of `WaitingFeeAcceptance`. 

> We could have cases where we won't be able to revert, just wanted to ensure we handle this case gracefully

That's why I added the new logic in `handle_amountless_update`. If we fail to revert right after trying to accept the fees, we'll detect this the next time we get `TransactionLockupFailed` and then revert. Do you see some edge cases I missed?